app/assets/javascripts/data_sets/editTable.js.coffee

```diff
@@ -17,148 +17,239 @@ uploadSettings =
   successEntry: (data, textStatus, jqXHR) ->
     window.location = data['displayURL']
 
-setupTable = (cols, data) ->
-  cols.push
-    id: 'del'
-    field: 'del'
-    name: ''
-    width: 0
-    formatter: (row, cell, value, columnDef, dataContext) ->
-      "<i class='fa fa-close slick-delete'></i>"
-
-  options =
-    autoEdit: true
-    editable: true
-    enableCellNavigation: true
-    enableColumnReorder: false
-    forceFitColumns: true
-    rowHeight: 35
-    syncColumnCellResize: true
-
-  for x, i in data
-    data[i]['id'] = i
-
-  view = new Slick.Data.DataView()
-  grid = new Slick.Grid '#slickgrid-container', view, cols, options
-  currId = data.length
-
-  view.setItems data
-
-  grid.onClick.subscribe (e, args) ->
-    cell = grid.getCellFromEvent e
-    if cell.cell == grid.getColumns().length - 1
-      grid.getEditorLock().commitCurrentEdit()
-      grid.resetActiveCell()
-      item = view.getItem cell.row
-      view.deleteItem item.id
-      grid.invalidate()
-
-      if grid.getDataLength() == 0
-        fields = {id: currId}
-        currId += 1
-        for x in cols
-          fields[x.field] = ''
-        view.addItem fields
-
-  view.onRowCountChanged.subscribe (e, args) ->
-    grid.updateRowCount()
-    grid.render()
-
-  view.onRowsChanged.subscribe (e, args) ->
-    grid.invalidateRow args.rows
-    grid.render()
-
-  $('.edit_table_add').click ->
-    if $('.edit_table_save').hasClass 'disabled'
-      return
-
-    grid.getEditorLock().commitCurrentEdit()
-    grid.resetActiveCell()
+Grid = (cols, data, submit) ->
```

Comment:
I maintain that it should be capital, but I will change it because I have picked my battles and this is not one of them.